### ISSUE TITLE
[Bug]: jdtls exits with code 13 and signal 0

### ISSUE THREAD

**Issue by cladera** [LunarVim/LunarVim#2090]
### Problem description

I wanted to give @ChristianChiarulli's [LunarVim config](https://github.com/ChristianChiarulli/lvim) a try so copied my current `~/.config/lvim` to a backup location and cloned Christian's config. Everything went ok, nothing apparently broke. However, the next time I rebooted my computer `jdtls` wouldn't start. The message I get is:

```
Client X quit with exit code 13 and signal 0
```

I assumed that the problem was caused by the config change so I restored my previous lvim folder but the problem persisted. I even uninstalled lvim and reinstalled it again but jdtls still fails.

**NOTE**: I don't see the folder `~/.config/lvim/ftplugin/` but I don't think I had one when jdtls was working. 

### Check the following

- [X] I am in a git managed directory
- [X] I restarted Neovim after running :LspInstall
- [X] I have a valid language configuration (~/.config/lvim/ftplugin/)
- [X] I checked the options (~/.config/lvim/config.lua)
- [X] The programs specified by the config are installed

### LunarVim version

dev-724420

### Neovim version (>= 0.5)

0.6.0

### Relevant log output

```shell
`:echo &ft`
java

`:LspInfo`
 

 Language client log: /Users/cgalancladera/.cache/nvim/lsp.log
 Detected filetype:   java
 
 0 client(s) attached to this buffer: 
 
 Other clients that match the filetype: java
 
 Config: jdtls
 	filetypes:         java
 	root directory:    [PROJECT_ROOT]
 	cmd:               /Users/cgalancladera/.sdkman/candidates/java/current/bin/java -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Dlog.protocol=true -Dlog.level=ALL -Xms1g -Xmx2G -javaagent:/Users/cgalancladera/.local/share/nvim/lsp_servers/jdtls/lombok.jar --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED -jar /Users/cgalancladera/.local/share/nvim/lsp_servers/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration /Users/cgalancladera/.local/share/nvim/lsp_servers/jdtls/config_mac -data /Users/cgalancladera/workspace/[PROJECT_BASE_PATH]
 	cmd is executable: true
 	autostart:         true
 	custom handlers:   language/status, workspace/applyEdit, textDocument/rename, textDocument/codeAction
 
 Configured servers list: null-ls, jdtls
```

`:checkhealth`

```
...
nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - ERROR: **Go**: not available
  - ERROR: **Composer**: not available
  - ERROR: **PHP**: not available
  - INFO: **bash**: `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)`
  - INFO: **tar**: `bsdtar 3.5.1 - libarchive 3.5.1 zlib/1.2.11 liblzma/5.0.5 bz2lib/1.0.8 `
  - INFO: **gzip**: `Apple gzip 352.40.6`
  - INFO: **curl**: `curl 7.77.0 (x86_64-apple-darwin21.0) libcurl/7.77.0 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.42.0`
  - INFO: **wget**: `GNU Wget 1.21.2 built on darwin21.1.0.`
  - INFO: **python3**: `Python 3.9.7`
  - INFO: **node**: `v17.0.1`
  - INFO: **Ruby**: `ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.x86_64-darwin21]`
  - INFO: **sh**: `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)`
  - INFO: **java**: `javac 11.0.9`
  - INFO: **RubyGem**: `3.0.3.1`
  - INFO: **pip3**: `pip 21.2.4 from /usr/local/lib/python3.9/site-packages/pip (python 3.9)`
  - INFO: **npm**: `8.1.0`
 ...
```

`LspStart jdtls`

```
Client X quit with exit code 13 and signal 0
```

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by cladera** [comment]
I forgot to mention that I tried cleaning lvim cache and clean & sync packer.

**Comment by cladera** [comment]
I managed to make it work by deleting the project workspace at `~/Workspace/[project_name]`. After that I've restarted nvim and jtdls started. 

Is there a command or instruction to rebuild the project or something similar for such cases?

**Comment by cladera** [comment]
I can confirm that messing around with lvim configuration and lsp servers obviously breaks jdtls workspace data. Removing the workspace and restarting lvim fixes such errors. 

I think this issue can be closed now, I'll open a new one explaining why I ended up messing my lvim configuration and what were my results.

**Comment by kylo252** [comment]
> I think this issue can be closed now, I'll open a new one explaining why I ended up messing my lvim configuration and what were my results.

Thank you! It would also be really helpful if you have some notes for this section here: https://www.lunarvim.org/languages/java.html#install-syntax-highlighting

**Comment by cladera** [comment]
@kylo252 I created a [repo](https://github.com/LunarVim/LunarVim/issues/2097) with my current lvim configuration that at least fixes this issues. However, as I explain here https://github.com/LunarVim/LunarVim/issues/2097, I've lost some code action features.

**Comment by abzcoding** [comment]
closed in favor of #2097

**Comment by punkucheung** [comment]
> 我忘了提到我尝试清理 lvim 缓存和清理和同步打包程序。

Client 2 quit with exit code 13 and signal 0 

This is also the case with mine. Has your problem been solved? How to solve it? Please tell me,

**Comment by punkucheung** [comment]
> I managed to make it work by deleting the project workspace at `~/Workspace/[project_name]`. After that I've restarted nvim and jtdls started.
> 
> Is there a command or instruction to rebuild the project or something similar for such cases?

Client 2 quit with exit code 13 and signal 0

This is also the case with mine. Has your problem been solved? How to solve it? Please tell me,

**Comment by punkucheung** [comment]
Language client log: /home/punku/.cache/nvim/lsp.log
 Detected filetype:   java
 
 0 client(s) attached to this buffer: 
 
 Other clients that match the filetype: java
 
 Config: jdtls
 	filetypes:         java
 	root directory:    /home/punku/workspace/demo
 	cmd:               /home/punku/.local/j/jdk11/bin/java -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -jar /home/punku/.local/share/nvim/lsp_servers/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration /home/punku/.local/share/nvim/lsp_servers/jdtls/config_linux -data /home/punku/workspace/workspace -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Djava.import.generatesMetadataFilesAtProjectRoot=false -Xmx1G -Xms100m
 	cmd is executable: true (checked by nvim-lsp-installer)
 	autostart:         true
 	custom handlers:   textDocument/codeAction, language/status, textDocument/rename, workspace/applyEdit
 
 Configured servers list: jdtls, jsonls, sumneko_lua

**Comment by lambdaydoty** [comment]
@punkucheung I can solve the same issue by cleaning up the cached file. Something like:
```
rm -rf $HOME/.local/share/jdtls/lc
```

**Comment by AxewBoTX** [comment]
> @punkucheung I can solve the same issue by cleaning up the cached file. Something like:
> 
> ```
> rm -rf $HOME/.local/share/jdtls/<project-name>
> ```

Im using neovim with jdtls-nvim on an Ubuntu machine from past 2 months and I got the same `client 0 exit with code 13` error and I tried all the solution that are given here, but none of them worked, when I tried this solution I got to know that I dont even have such folder in my ~/.local/share folder.

Is this issue solved, or like can you tell me how to fix this?

**Comment by crhf** [comment]
@AxewBoTX it's the `-data` option if you are using https://github.com/mfussenegger/nvim-jdtls#configuration-verbose.

**Comment by Karmenzind** [comment]
Met the same error under neovim with default lsp config. `rm -rf ~/.cache/jdtls` worked for me.

**Comment by idelice** [comment]
> Met the same error under neovim with default lsp config. `rm -rf ~/.cache/jdtls` worked for me.

This fixed it for me too, but instead of deleting the parent folder, I've only deleted the project that were having this issue!

**Comment by blagoySimandov** [comment]
I had to first set `$JAVA_HOME` ( I think I had two java version installed and one was incompatible with jdtls.)  After that cleared the cache as shown above and everything seems to be working now.

**Comment by alvaroarenas** [comment]
I had the same problem described here and I am leaving my "issue" for visitors from the future. 

TL;DR; 
1. Make sure you have the compatible java version (in my case java 21)
2. Delete your jdtls cache. 

I had unfortunately hardcoded the JAVA_HOME in my newly adopted fishshell configuration. I am using sdkman for setting the default java but of course it was never picked because I overwritten it  🤦🏻

**Comment by TheBotlyNoob** [comment]
For me, the jdtls cache was in `~/.cache/nvim/jdtls`.

**Comment by joseiedo** [comment]
This happens so often that I had to create a command lol

```lua
vim.api.nvim_create_user_command(
  'CleanJdtls',
  function()
    vim.fn.delete(vim.fn.expand('~/.cache/nvim/jdtls'), 'rf')
    vim.fn.delete(vim.fn.expand('~/.cache/jdtls'), 'rf')
    vim.fn.delete(vim.fn.expand('~/.local/share/jdtls'), 'rf')
    print('Deleted cache!')
  end,
  {}
)
```

**Comment by co-odw** [comment]
If you're on linux this worked for me. You can confirm that they exist in the first place, before removing.
```rm -rf /tmp/jdtls-*```

**Comment by nkromm** [comment]
> Make sure you have the compatible java version (in my case java 21)

This is what resolved this problem for me. I am using java 17 so I had to downgrade to jdtls 1.43.0 (https://download.eclipse.org/jdtls/milestones/1.43.0/) which seems to be the last version that supports java 17.

They say here, in the pr that changed the minimum requirement, (https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/3358#issuecomment-2729368625) that this is not necessary and you can just run the jdtls on jre 21+ and have your project in java 17, but I did not try it. Downgrading was good enough for now.

Leaving this comment in case it helps someone like me.
